packages/design-picker/src/components/unified-design-picker.tsx

```diff
@@ -216,21 +217,40 @@ const DesignCardGroup = ( {
 }: DesignCardGroup ) => {
 	const translate = useTranslate();
 	const [ isCollapsed, setIsCollapsed ] = useState( !! categoryName || false );
+	const { selectedCategories } = useDesignPickerFilters();
+
+	const shuffleSeed = useMemo(
+		() =>
+			selectedCategories
+				.join( '' )
+				.split( '' )
+				.reduce( ( acc, char ) => acc + char.charCodeAt( 0 ), 0 ),
+		[ selectedCategories ]
+	);
 
 	const visibleDesigns = useMemo( () => {
 		const free = designs.filter( ( design ) => design.design_tier === FREE_THEME );
 		const boosted = free.slice( 0, FREE_DESIGNS_BOOSTED_COUNT );
 		const remaining = designs.filter( ( design ) => ! boosted.includes( design ) );
 
-		if ( ! isCollapsed ) {
-			return [ ...boosted, ...remaining ];
+		// Ensure Best Matching Themes changes whenever selected categories are updated.
+		// This provides visual feedback so that users notice that the results have changed.
+		if ( category === 'best' ) {
+			return [
+				...shuffleDesigns( boosted, shuffleSeed ),
+				...shuffleDesigns( remaining, shuffleSeed ),
```

Comment:
During my testing, when `boosted.length` = `remaining.length` = 3, it could be the case that `shuffleDesign()` shuffles the 3-element array into the original ordering (~ 1/6 chance). In this case, `boosted` and `remaining` will both remain the same and the issue is not solved.

Maybe it's better if the seed is different between them? E.g. something like

```suggestion
				...shuffleDesigns( boosted, shuffleSeed ),
				...shuffleDesigns( remaining, shuffleSeed + 1 ),
```

Reply:
I used a prime number because they seemed to be better with distribution https://computinglife.wordpress.com/2008/11/20/why-do-hash-functions-use-prime-numbers/